src/dmd/dmodule.d

```diff
@@ -405,39 +405,51 @@ extern (C++) final class Module : Package
     size_t nameoffset;          // offset of module name from start of ModuleInfo
     size_t namelen;             // length of module name in characters
 
-    extern (D) this(const ref Loc loc, const(char)* filename, Identifier ident, int doDocComment, int doHdrGen)
+    extern (D) this(const ref Loc loc, const(char)[] filename, Identifier ident, int doDocComment, int doHdrGen)
     {
         super(loc, ident);
-        const(char)* srcfilename;
+        const(char)[] srcfilename;
         //printf("Module::Module(filename = '%s', ident = '%s')\n", filename, ident.toChars());
         this.arg = filename;
-        srcfilename = FileName.defaultExt(filename, global.mars_ext);
-        if (global.run_noext && global.params.run && !FileName.ext(filename) && FileName.exists(srcfilename) == 0 && FileName.exists(filename) == 1)
+        srcfilename = FileName.defaultExt(filename, global.mars_ext.toDString());
```

Comment:
I see all these `toDString()` calls everywhere, which do a `strlen()`. Not sure this is an improvement. Also, `mars_ext` should be declared as a string, and avoid the `toDString()`s.

Reply:
Until all C strings have been changed to D strings I think we have to live with `toDString` and some inefficiencies like calling `strlen`.

Reply:
As I mentioned in another comment, defaultExt and SetOutFile immediately call toDstring on everything for the char* overload, so we're actually decreasing the net strlen calls here.  I'll continue to slicify things to remove those.